sphinx_rtd_theme/layout.html

```diff
@@ -41,11 +41,13 @@
   {%- endfor -%}
 
   {#- FAVICON #}
-  {%- if favicon %}
-    {%- if sphinx_version_info < (4, 0) -%}
-    <link rel="shortcut icon" href="{{ pathto('_static/' + favicon, 1) }}"/>
-    {%- else %}
-    <link rel="shortcut icon" href="{{ favicon_url }}"/>
+  {%- if sphinx_version_info < (4, 0) -%}
+    {%- if favicon %}
+      <link rel="shortcut icon" href="{{ pathto('_static/' + favicon, 1) }}"/>
+    {%- endif %}
+  {%- else %}
+    {%- if favicon_url %}
+      <link rel="shortcut icon" href="{{ favicon_url }}"/>
```

Comment:
This seems to miss the conditional `sphinx_version_info > (4,0) and favicon`. Similar to below, nested conditionals here can be reduce easily:

```suggestion
  {% set _favicon_url = favicon_url | default:pathto('_static/' + favicon, 1) %}
  <link rel="shortcut icon" href="{{ _favicon_url }}"/>
```

Reply:
Same answer - I can change it to at the loss of the version logic (which makes it easier to remove in the future) and then test it again.

Your call :+1:

Reply:
The version conditional seems to be causing more problems than it's solving, so I think it's fair to remove this.

We really only care about the variables being used, the Sphinx version doesn't help us in this determination.

Reply:
Alright, that's up to you, I'll re-implement it with the suggestions that you gave and re-test it :+1: